exercises/mongo-shell/intermediate-read.md

```diff
@@ -0,0 +1,146 @@
+## Comparison Operators
+
+These operators allow us to match a field's value relative to another value. 
```

Comment:
I think we should explain what **Comparison Operators** are.  Remember that these students are very new to this stuff, having only completed undergrad, so it's helpful to see this type of thing as a review for them.

Additionally, each language seems to have it's own operators, so a quick overview of what MongoDB has isn't a bad thing.

Lastly, MongoDB has **two** set's of **Comparison Operators**, one set for normal queries, and another for aggregate queries.  We need to make a distinction here somehow.

```suggestion
Comparison operators are used in conditions that compare one expression with another, typically containing a boolean return value.  

The MongoDB query language is unique because it uses **comparison Operators** in two different ways:  
- Comparison Query Operators
- Comparison Aggregation Operators

Let's take a quick :eyes: at what is available under each context before moving forward.

**Comparison Query Operators**
|Name|Description|
|---|---|
|$eq|Matches values that are equal to a specified value.|
|$gt | Matches values that are greater than a specified value.|
|$gte | Matches values that are greater than or equal to a specified value.|
|$in | Matches any of the values specified in an array.|
|$lt | Matches values that are less than a specified value.|
|$lte | Matches values that are less than or equal to a specified value.|
|$ne | Matches all values that are not equal to a specified value.|
|$nin | Matches none of the values specified in an array.|

**Comparison Aggregation Operators**
|Name|Description|
|---|---|
|$eq|Returns **true** if the values are equivalent.|
|$gt | Returns **true** if the first value is greater than the second.|
|$gte |Returns **true** if the first value is greater than or equal to the second.|
|$lt | Returns **true** if the first value is less than the second.|
|$lte | Returns **true** if the first value is less than or equal to the second.|
|$ne | Returns **true** if the value are *not* equivalent.|
|$cmp | Returns: 0 if the two values are equivalent, 1 if the first value is greater than the second, and -1 if the first value is less than the second.|

`$cmp` is the only **Comparison Aggregation Operator** that does not return a boolean!
```

Reply:
This document pertains to read operations at an intermediate level. I figured I'd put the aggregation pipeline in the advanced read document. 

```suggestion

Comparison operators are used in conditions that compare one expression with another, typically containing a boolean return value.  

The MongoDB query language is unique because it uses **comparison Operators** in two different ways:  
- Comparison Query Operators
- Comparison Aggregation Operators

Let's take a quick :eyes: at what is available for **Query Selector** operations.

**Comparison Query Operators**
|Name|Description|
|---|---|
|$eq|Matches values that are equal to a specified value.|
|$gt | Matches values that are greater than a specified value.|
|$gte | Matches values that are greater than or equal to a specified value.|
|$in | Matches any of the values specified in an array.|
|$lt | Matches values that are less than a specified value.|
|$lte | Matches values that are less than or equal to a specified value.|
|$ne | Matches all values that are not equal to a specified value.|
|$nin | Matches none of the values specified in an array.|
```